plugins/postcss-cascade-layers/src/index.ts

```diff
@@ -82,9 +83,11 @@ const creator: PluginCreator<undefined> = () => {
 			//  - move out styles from atRule, insert before: https://postcss.org/api/#container-insertbefore
 			//  - delete empty atRule
 			//  - give selectors the specificity they need based on layerPriority state
-			// root.walkAtRules((atRule) => {
-			// 	console.log(atRule, "third walkthrough");
-			// });
+			root.walkAtRules('layer', (atRule) => {
+				// move out styles from atRule, insert before
+
+				atRule.replaceWith(new Rule({nodes: atRule.nodes, source: atRule.source, selector: `${generateNot(layerOrder[atRule.params])}`}));
```

Comment:
@romainemenke 

I know you said to use `insertbefore`, but how do i control the selector with that option? Couldn't figure it out.

Reply:
@oluoluoxenfree could we do something like this with `insertBefore` ? 
```
atRule.each((node) => {
    const transformedNode = "p:not(#\#) { max-width: 70ch; }" // use generateNot to get selector + specificity
    root.insertBefore(atRule, transformedNode);
});
atRule.remove()
```

Reply:
did you get any errors around it needing a source locally? 

I rewrote it to not manually create a rule, but getting different errors. 

Is what I have now along the lines of what you had in mind?